### PR TITLE
Add 'action' response in modify_contract

### PR DESCRIPTION
**Pull request by brashidnia-figure**
- Adding metadata on the `Response` to be able to easily discern the `TxEvent` type from the EventStream.
- Bumped version to `0.19.2`

### CODE REVIEW

**Review by jtalis — DISMISSED** [review]

**Review by leeduan — APPROVED** [review]

File: .github/workflows/rust.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
       - name: cargo tarpaulin xml report
         uses: actions-rs/tarpaulin@v0.1
         with:
-          version: '0.19.1'
```
Comment (brashidnia-figure) [inline .github/workflows/rust.yaml]: This is completely unrelated haha, this is the `cargo-tarpaulin` version for the GHA flow. I just upgraded to the latest

File: src/contract.rs [hunk 2]
```diff
@@ -1120,7 +1120,7 @@ fn modify_contract(
         bid_required_attributes,
     )?;
 
-    let response = Response::new();
+    let response = Response::new().add_attribute("action", "modify_contract");
```
Comment (mwoods-figure) [inline src/contract.rs:1123]: Are there potentially other `"action"` values that could be put there? It might be worthwhile to define some kind of enum with these actions as variants and convert it to a string when building `Response`.

**Review by mwoods-figure — APPROVED** [review]

File: src/contract.rs (hunk 2 above)
Comment (brashidnia-figure) [reply src/contract.rs:1123]: i think we could use an enum, right now we are using `action` in the response to easily identity the type of contract execution that took place when consuming blockchain data. `create_ask`, `create_bid`, ... etc. all have their own `action` defined via a hardcoded string right now (could be converted to an enum)

I could try to update them all

File: src/contract.rs [hunk 3]
```diff
@@ -124,7 +124,7 @@ pub fn instantiate(
             "contract_info",
             format!("{:?}", get_contract_info(deps.storage)?),
         ),
-        attr("action", "init"),
+        attr("action", ContractAction::Init.to_string()),
```
Comment (leeduan) [inline src/contract.rs:127]: does this change keep the existing casing?

Comment (brashidnia-figure) [reply src/contract.rs:127]: ya, i'll double check, but the unit tests do check the result against an expected string. I believe the Serde snake_case property on the enum makes it match the existing case

Comment (brashidnia-figure) [reply src/contract.rs:127]: confirmed its snake cased

**Review by mwoods-figure — APPROVED** [review]

**Review by leeduan — APPROVED** [review]
